roofit/roofitcore/src/RooHistFunc.cxx

```diff
@@ -200,6 +200,49 @@ Double_t RooHistFunc::evaluate() const
   return ret ;
 }
 
+
+////////////////////////////////////////////////////////////////////////////////
+/// Compute value of the HistFunc for every entry in `evalData`.
+/// \param[in/out] evalData Struct with input data. The computation results will be stored here.
+/// \param[in] normSet Set of observables to normalise over (ignored).
+RooSpan<double> RooHistFunc::evaluateSpan(RooBatchCompute::RunContext& evalData, const RooArgSet* /*normSet*/) const {
+  std::vector<RooSpan<const double>> inputValues;
+  std::size_t batchSize = 0;
+  for (const auto& obs : _depList) {
+    auto realObs = dynamic_cast<const RooAbsReal*>(obs);
+    if (realObs) {
+      auto inputs = realObs->getValues(evalData, nullptr);
+      batchSize = std::max(batchSize, inputs.size());
+      inputValues.push_back(std::move(inputs));
+    } else {
+      inputValues.emplace_back();
+    }
+  }
+
+  auto results = evalData.makeBatch(this, batchSize);
+
+  for (std::size_t i = 0; i < batchSize; ++i) {
+    bool skip = false;
+
+    for (auto j = 0u; j < _histObsList.size(); ++j) {
+      const auto histObs = _histObsList[j];
+
+      if (i < inputValues[j].size()) {
+        histObs->setCachedValue(inputValues[j][i], false);
+        if (!histObs->inRange(nullptr)) {
+          skip = true;
+          break;
+        }
+      }
+    }
+
+    results[i] = skip ? 0. : _dataHist->weight(_histObsList, _intOrder, false, _cdfBoundaries);
```

Comment:
That's nice. I would actually suggest to put an assert into that function that checks that the preconditions are met. I costs minimal run time in debug builds, but it will help a lot in debugging if you violate the preconditions.
I'll add such a commit.

Reply:
I take that back, because it's already happening.

Reply:
Actually no. It was only checking the size. Commit coming.